### DISCUSSION TITLE
Problem with Lora32 v3 from Heltec

### DISCUSSION THREAD

**Discussion by AnnaPiaIannone** [LoRaMesher/LoRaMesher#56] [Q&A] [answered]
Hi, I'm using the Heltec Lora 32 v3 and I noticed that the start() function generates this problem that blocks the entire code:
E (15129) gpio: gpio_isr_handler_remove(480): GPIO isr service is not installed, call gpio_install_isr_service() first.
I tried installing it in setup with 
"esp_err_t err = gpio_install_isr_service(ESP_INTR_FLAG_LEVEL1);
 if (err == ESP_ERR_INVALID_STATE) {
 Serial.println("ISR service already installed");
 } else if (err != ESP_OK) {
 Serial.printf("Error installing ISR service: %d\n", err);
 }",
 but then it generates this "E (15129) gpio: gpio_install_isr_service(449): GPIO isr service already installed".
Thanks in advance for the help.

PS. I say it is the start() function because I tried to remove it and ONLY the transmission works

**Comment by stan69b** [comment]
I'm running into the same probleme with Heltec v3 lora. tried fiddling with the code but could not get it to work ether :/ not much on the internet about this, it says it is not started but if you start it it says it is already started so debugging is not easy. 
I would also like a answer/fix for this :) 
Answering so this issue gets a little more visibility

**Reply by stan69b** [reply]
issue related to this :
https://github.com/LoRaMesher/LoRaMesher/issues/57

**Reply by Jaimi5** [reply]
Hi!
Sorry for the delay—I’m working on reproducing the error and hope to identify it soon.
As for the "E (15129) gpio: gpio_install_isr_service(449): GPIO ISR service already installed" error in my other projects, it doesn’t seem to interfere with the program's execution.
Additionally, here are the pins that I am currently using:

```
#define LORA_SCK 5
#define LORA_MISO 19
#define LORA_MOSI 27
#define LORA_CS 18
#define LORA_RST 14
#define LORA_IRQ 26
#define LORA_IO1 33

SPIClass newSPI(HSPI);

void setup() {
    LoraMesher::LoraMesherConfig config = LoraMesher::LoraMesherConfig();

    config.loraCs = LORA_CS;
    config.loraRst = LORA_RST;
    config.loraIrq = LORA_IRQ;
    config.loraIo1 = LORA_IO1;

    config.module = LoraMesher::LoraModules::SX1276_MOD;

    newSPI.begin(LORA_SCK, LORA_MISO, LORA_MOSI, LORA_CS);
    config.spi = &newSPI;

    radio.begin(config);

    //Create the receive task and add it to the LoRaMesher
    createReceiveMessages();

    //Start LoRaMesher
    radio.start();

....

```

Additionally, RadioLib, which is the base that we use to communicate with the LoRa chip, uses the following initialization:

gpio_install_isr_service((int)ESP_INTR_FLAG_IRAM);

Hope this helps.

**Reply by stan69b** [reply, marked as answer]
Thanks, i tried using platform io (which you recommended so…..) and it works. Although the display does not seem to display anything. It works as intended on my lilygo.
I guess there is a problem when trying it in arduono ide but again, i think the repo says it is ment for platform IO so maybe it is not a bug persay. 
anyway, the project is very cool, looking forward to implementing it properly on multiple devices and get my system up and running. Thanks for all the work ☺️
